### PR TITLE
Docs cleanup

### PR DESCRIPTION
**Pull request by aclark4life**
One final point of confusion to address: AFAICT, references were part of the PIL Handbook? But I left them at the top level of the documentation because that seems more appropriate. I'm also slightly confused and/or can't recall how API documentation is generated using Sphinx (one time vs. each time?) In any event, here's a big cleanup for 3.0

### CODE REVIEW

**Comment by hugovk** [comment]
RTD builds the docs for every commit:
https://readthedocs.org/projects/pillow/builds/

File: docs/developer/index.rst [hunk 1]
```diff
@@ -15,3 +14,17 @@ Version number
 The version number is currently stored in 3 places::
 
     PIL/__init__.py _imaging.c setup.py
+
+
+Readthedocs
+~~~~~~~~~~~
+
+Make sure the default version for Readthedocs is the latest release version e.g. 3.0.0 not latest.
```
Comment (hugovk) [inline docs/developer/index.rst]: "Readthedocs" -> "Read the Docs". Also as a developer, and as a collaborator, I've no idea where this version is.

File: docs/developer/index.rst [hunk 2]
```diff
@@ -3,7 +3,6 @@ Developer
```
Comment (hugovk) [inline docs/developer/index.rst]: I'm not sure we need this file. We have https://github.com/python-pillow/Pillow/blob/master/CONTRIBUTING.md for general things a contributor should check, whether bugs or code. A good thing about that file is it's autolinked from the "Please review the guidelines for contributing to this repository." banner on https://github.com/python-pillow/Pillow/issues/new and when creating PRs. (The CI skip note can go in there.)

For releasers, we have https://github.com/python-pillow/Pillow/blob/master/RELEASING.md

File: docs/developer/index.rst (hunk 1 above)
Comment (aclark4life) [inline docs/developer/index.rst]: No idea where what version is? You mean RTD administrative tasks?

**Comment by aclark4life** [comment]
No I mean if you start a new project with lots of Python code and no restructured text, how do you produce an API reference? I can't recall…

File: docs/developer/index.rst (hunk 1 above)
Comment (hugovk) [inline docs/developer/index.rst]: Where the default version for Read the Docs is actually set. Perhaps link to the page where this setting is.

File: docs/developer/index.rst (hunk 2 above)
Comment (aclark4life) [inline docs/developer/index.rst]: Gone! Thanks.

**Comment by wiredfool** [comment]
Would it make sense to build the docs in testing, to make sure that they're going to build properly?   Not that we've had that sort of failure recently, but it's has happened in the past.

File: docs/developer/index.rst (hunk 1 above)
Comment (aclark4life) [inline docs/developer/index.rst]: K, moved to RELEASING & added link to https://readthedocs.org/dashboard/pillow/versions/

**Comment by aclark4life** [comment]
@wiredfool You mean CI testing? Sure, probably can't hurt.

**Comment by wiredfool** [comment]
Yep, in travis. It would be great to break on new warnings, but I'd settle for just making sure that they build and don't fail.

**Comment by aclark4life** [comment]
@wiredfool Done (hopefully correctly in .travis.yml)

**Comment by hugovk** [comment]
```
$ cd docs; make html; make linkcheck
sphinx-build -b html -d _build/doctrees   . _build/html
make: sphinx-build: Command not found
make: *** [html] Error 127
sphinx-build -b linkcheck -d _build/doctrees   . _build/linkcheck
make: sphinx-build: Command not found
make: *** [linkcheck] Error 127
```

Looks like sphinx needs installing too.

Would be good to do `pushd docs; make html; make linkcheck; popd` to return back to the original directory afterwards. 

Once it's working, could also create a throwaway branch from this and break some docs to confirm it fails a build.

**Comment by aclark4life** [comment]
Any reason not to run pip install -r requirements.txt ?

**Comment by wiredfool** [comment]
There should be a `make docs` target in the main make file that makes the html docs in the subdirectory. Might be easier than mucking around with pushd/popd

**Comment by aclark4life** [comment]
Well, a ``make docs`` could certainly automate pushd/popd but probably can wait until after this is merged.

**Comment by wiredfool** [comment]
Actually, it's `make doc`: https://github.com/python-pillow/Pillow/blob/master/Makefile#L20

**Pull request merged**